### PR TITLE
Alerts can be shared via a shareable URL #3184

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Fixes https://github.com/elastic/security-docs/issues/3184.

Previews:
- View detection alert details: Added explanation of how to use the Share alert button and the new `kibana.alert.url` field. Also refreshed screenshot.
- Alert schema: Added the new `kibana.alert.url` field.

### CODE REVIEW

File: docs/detections/alerts-view-details.asciidoc [hunk 1]
```diff
@@ -18,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+To share an alert, click the **Share alert** link in the upper right corner of the flyout. This gives you a shareable URL that you can share with others or paste into a browser.  
+
+IMPORTANT: The **Share alert** link gives you a direct link to the alert's details flyout. Do _not_ use the general URL in the browser to share the alert. That URL will not work.
```
Comment (michaelolo24) [inline docs/detections/alerts-view-details.asciidoc:19]: I would rephrase this to something like:

> The Share alert link gives you a direct link to the alert’s details flyout. Using the general URL in the browser to share the alert can cause inconsistencies when sharing due to relative date ranges and filters that you may currently have set in your browser. The share button is the most reliable way to confirm the alert you want to be shared will be shared with the user without the inconsistencies of the other filters.

This also needs to be added:
> Also, if grouping is enabled when navigating to an alert url, the group must be expanded for the flyout to auto-expand

File: docs/detections/alerts-view-details.asciidoc [hunk 2]
```diff
@@ -18,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+To share an alert, click the **Share alert** link in the upper right corner of the flyout. This gives you a shareable URL that you can share with others or paste into a browser.  
+
+IMPORTANT: The **Share alert** link gives you a direct link to the alert's details flyout. Do _not_ use the general URL in the browser to share the alert. That URL will not work. 
+
+NOTE: If you've set the {kibana-ref}/settings.html#server-publicBaseUrl[`server.publicBaseUrl`] configuration setting in the `kibana.yml` file, you can also find the shareable URL in the `kibana.alert.url` field. You can find the field by going the *Table* tab and searching for `kibana.alert.url`.
```
Comment (michaelolo24) [inline docs/detections/alerts-view-details.asciidoc:21]: 👌🏾

**Comment by michaelolo24** [comment]
Only other thing is I think that needs to be done is updating the screenshot 😄

File: docs/detections/alerts-view-details.asciidoc (hunk 1 above)
Comment (nastasha-solomon) [reply docs/detections/alerts-view-details.asciidoc]: Incorporated feedback in [ce95403](https://github.com/elastic/security-docs/pull/3264/commits/ce954031c28b63d971e0e874f7d40a7a8e8af3c2),

File: docs/detections/alerts-view-details.asciidoc [hunk 3]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a direct way to share alert details. Click the link to get a shareable URL that you can paste into browsers, cases, messages, and more. Do _not_ use the general browser URL to share the alert. It can cause inconsistent results if the Alerts page has filters or relative time ranges.
```
Comment (nastasha-solomon) [inline docs/detections/alerts-view-details.asciidoc:17]:
```suggestion
The **Share alert** link gives you a direct way to share alert details. Click the link to get a shareable URL that you can paste into browsers, cases, messages, and more. We don't recommend using the general browser URL to share the alert. It can cause inconsistent results if the Alerts page has filters or relative time ranges. 
```

**Review by michaelolo24 — APPROVED** [review]
Looks great, thanks Nastasha! LGTM 🚀

File: docs/detections/alerts-view-details.asciidoc [hunk 4]
```diff
@@ -1,10 +1,10 @@
 [[view-alert-details]]
 == View detection alert details
 
-To inspect a detection alert, click the *View details* button from the Alerts table. The alert details flyout appears with several options to view alert data.
+To inspect a detection alert, click the *View details* button from the Alerts table. The alert details flyout appears with a way to share the alert and several options to view alert data.
```
Comment (joepeeples) [inline docs/detections/alerts-view-details.asciidoc:4]: I'm not sure this is necessary to call out in the intro paragraph — it's a cool feature, but probably not the main reason users will be opening the alert details flyout. We explain the feature in detail almost immediately below, which seems sufficient.

File: docs/detections/alerts-view-details.asciidoc [hunk 5]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a direct way to share alert details. Click the link to get a shareable URL that you can paste into browsers, cases, messages, and more. We don't recommend using the general browser URL to share the alert. It can cause inconsistent results if the Alerts page has filters or relative time ranges. 
+
+IMPORTANT: If you've enabled grouping on the Alerts page, the alert details flyout won't open until you expand the collapsed group. 
+
+NOTE: If you've set the {kibana-ref}/settings.html#server-publicBaseUrl[`server.publicBaseUrl`] configuration setting in the `kibana.yml` file, you can also find the shareable URL in the `kibana.alert.url` field. You can find the field by going the *Table* tab and searching for `kibana.alert.url`.
```
Comment (joepeeples) [inline docs/detections/alerts-view-details.asciidoc:21]: A few nit edit suggestions for wordiness, repetition, etc. Also, "direct way" and "general browser URL" weren't super clear to me, so added some tweaks.
```suggestion
The **Share alert** link gives you a way to share alert details. Click the link to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges. 

IMPORTANT: If you've enabled grouping on the Alerts page, the alert details flyout won't open until you expand the collapsed group. 

NOTE: If you've set the {kibana-ref}/settings.html#server-publicBaseUrl[`server.publicBaseUrl`] configuration setting in the `kibana.yml` file, the shareable URL is also in the `kibana.alert.url` field. You can find the field by searching for `kibana.alert.url` on the *Table* tab.
```

**Review by joepeeples — COMMENTED** [review]
Looks good, added a few nits. Thanks!

File: docs/detections/alerts-view-details.asciidoc [hunk 6]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a direct way to share alert details. Click the link to get a shareable URL that you can paste into browsers, cases, messages, and more. We don't recommend using the general browser URL to share the alert. It can cause inconsistent results if the Alerts page has filters or relative time ranges.
```
Comment (benironside) [inline docs/detections/alerts-view-details.asciidoc:17]:
```suggestion
The **Share alert** link gives you a direct way to share alert details. Click it to get a URL that you can paste into browsers, cases, messages, and more. We don't recommend using the general browser URL to share the alert. It can cause inconsistent results if the Alerts page has filters or relative time ranges. 
```

**Review by benironside — COMMENTED** [review]
Looks good! Didn't find much to add to Joe's review.

File: docs/detections/alerts-view-details.asciidoc (hunk 4 above)
Comment (nastasha-solomon) [reply docs/detections/alerts-view-details.asciidoc:4]: That's fair. The main purpose and value of the flyout is to shore more info on the alert. I can remove this bit.

```suggestion
To inspect a detection alert, click the *View details* button from the Alerts table. The alert details flyout appears with several options to view alert data.
```

File: docs/detections/alerts-view-details.asciidoc [hunk 7]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a way to share alert details. Click the link to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges.
```
Comment (nastasha-solomon) [inline docs/detections/alerts-view-details.asciidoc:17]:
```suggestion
The **Share alert** link gives you a way to share alert details. Click it to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges. 
```

File: docs/detections/alerts-view-details.asciidoc [hunk 8]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a way to share alert details. Click it to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges.
```
Comment (jmikell821) [inline docs/detections/alerts-view-details.asciidoc:17]:
```suggestion
The **Share alert** link allows you to share alert details. Click it to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges. 
```

File: docs/detections/alerts-view-details.asciidoc [hunk 9]
```diff
@@ -14,6 +14,12 @@ The alert details flyout contains these informational tabs:
 * *JSON*: The alert data in JSON format.
 * *Osquery Results*: Results from queries attached to rules display on the *Osquery Results* tab. <<add-osquery-response-action,Learn more.>>
 
+The **Share alert** link gives you a way to share alert details. Click it to copy a URL that you can paste into browsers, cases, messages, and more. We recommend doing this instead of copying the URL from your browser's address bar, which can cause inconsistent results if the Alerts page has filters or relative time ranges. 
+
+IMPORTANT: If you've enabled grouping on the Alerts page, the alert details flyout won't open until you expand the collapsed group. 
+
+NOTE: If you've set the {kibana-ref}/settings.html#server-publicBaseUrl[`server.publicBaseUrl`] configuration setting in the `kibana.yml` file, the shareable URL is also in the `kibana.alert.url` field. You can find the field by searching for `kibana.alert.url` on the *Table* tab.
```
Comment (jmikell821) [inline docs/detections/alerts-view-details.asciidoc:21]: Was playing with the language here; totally optional. 
```suggestion
NOTE: If you've configured the {kibana-ref}/settings.html#server-publicBaseUrl[`server.publicBaseUrl`] setting in the `kibana.yml` file, the shareable URL is also in the `kibana.alert.url` field. You can find the field by searching for `kibana.alert.url` on the *Table* tab.
```

**Review by jmikell821 — COMMENTED** [review]
Slight suggestion but looks great, thanks!
